avax-synapse-stablepool/contracts/Strategy.sol

```diff
@@ -0,0 +1,484 @@
+            }
+
+            swapPool.removeLiquidityOneToken(toWithdraw, curveId, 0, now);
+        }
+
+        uint256 diff = want.balanceOf(address(this)).sub(wantBalanceBefore);
+
+        if (diff > _amount) {
+            _liquidatedAmount = _amount;
+        } else {
+            _liquidatedAmount = diff;
+            _loss = _amount.sub(diff);
+        }
+    }
+
+    function liquidatePosition(uint256 _amountNeeded)
+        internal
+        override
+        returns (uint256 _liquidatedAmount, uint256 _loss)
+    {
+        uint256 wantBal = want.balanceOf(address(this));
+
+        if (wantBal < _amountNeeded) {
+            (_liquidatedAmount, _loss) = withdrawSome(_amountNeeded.sub(wantBal));
+        }
+
+        _liquidatedAmount = Math.min(_amountNeeded, _liquidatedAmount.add(wantBal));
+    }
+
+    function liquidateAllPositions() internal override returns (uint256 _amountFreed) {
+        (_amountFreed, ) = liquidatePosition(1e36); //we can request a lot. dont use max because of overflow
+    }
+
+    function adjustPosition(uint256 _debtOutstanding) internal override {
+        if (lastInvest.add(minTimePerInvest) > block.timestamp) {
+            return;
+        }
+
+        uint256 _wantToInvest = Math.min(want.balanceOf(address(this)), maxSingleInvest);
+        if (_wantToInvest == 0) {
+            return;
+        }
+
+        uint256 expectedOut = _wantToInvest.mul(1e18).div(virtualPriceToWant());
+        uint256 maxSlip = expectedOut.mul(DENOMINATOR.sub(slippageProtectionIn)).div(DENOMINATOR);
+
+        if (isWantWETH()) {
+            weth.withdraw(_wantToInvest);
+            uint256[2] memory amounts;
+            amounts[0] = _wantToInvest;
+            swapPool.addLiquidity{value: _wantToInvest}(amounts, maxSlip, now);
+        } else {
+            uint256[] memory amounts = new uint256[](poolSize);
+            amounts[curveId] = _wantToInvest;
+            swapPool.addLiquidity(amounts, maxSlip, now);
+        }
+        // deposit lpToken 
+        yvToken.deposit();
+        lastInvest = block.timestamp;
+    }
+
+    function prepareMigration(address _strategy) internal override {
+        yvToken.transfer(_strategy, yvToken.balanceOf(address(this)));
+    }
+
+    function protectedTokens()
+        internal
+        override
+        view
+        returns (address[] memory) {
+
+        address[] memory protected = new address[](1);
+          protected[0] = address(yvToken);
+
+          return protected;
+    }
+
+    /* ========== SETTERS ========== */
+
+    ///@notice This allows us to manually harvest with our keeper as needed
+    function setForceHarvestTriggerOnce(bool _forceHarvestTriggerOnce)
+        external
+        onlyAuthorized
+    {
+        forceHarvestTriggerOnce = _forceHarvestTriggerOnce;
+    }
+
+    ///@notice When our strategy has this much credit, harvestTrigger will be true.
+    function setMinHarvestCredit(uint256 _minHarvestCredit)
```

Comment:
No need for this mechanism, we harvest on a time basis, this can make for more expensive bot cost